### PR TITLE
Fix annotation translation when the value is 0

### PR DESCRIPTION
**Pull request by pauzinl**
This patch fixes translation for annotations, when all values of annotation
are 0.

### CODE REVIEW

**Comment by pauzinl** [comment]
@MrSidims @KornevNikita take a look please

File: lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -2579,6 +2579,17 @@ void processOptionalAnnotationInfo(Constant *Const,
         AnnotationString += std::to_string(CInt->getSExtValue());
       }
     }
+  } else if (auto *ZeroStruct =
+                 dyn_cast<ConstantAggregateZero>(Const->getOperand(0))) {
+    // It covers case when when all elements of struct are 0 and they become
```
Comment (KornevNikita) [inline lib/SPIRV/SPIRVWriter.cpp:2584]: Typo

File: test/transcoding/annotate_attribute.ll [hunk 2]
```diff
@@ -44,8 +47,10 @@ target triple = "spir64-unknown-linux"
 @.str.5 = private unnamed_addr constant [4 x i8] c"128\00", section "llvm.metadata"
 @.str.6 = private unnamed_addr constant [4 x i8] c"qux\00", section "llvm.metadata"
 @.str.7 = private unnamed_addr constant [22 x i8] c"my_custom_annotations\00", section "llvm.metadata"
+@.str.8 = private unnamed_addr constant [31 x i8] c"annotation_with_zerointializer\00", section "llvm.metadata"
 @.args.0 = private unnamed_addr constant { i32, i32 } { i32 30, i32 60 }, section "llvm.metadata"
 @.args.1 = private unnamed_addr constant { i32, i32, i32 } { i32 20, i32 60, i32 80 }, section "llvm.metadata"
+@.args.2 = private unnamed_addr constant { i32, i32, i32 } zeroinitializer, section "llvm.metadata"
```
Comment (KornevNikita) [inline test/transcoding/annotate_attribute.ll:53]: Could you also add a check if the value is `false` (`[[clang::annotate("my_annotation", false)]]`). And add a case for a mix with 0 e.g. `{ i32 0, i32 1, i32 0 }` to make sure that the issue occures only if all values are zero?

File: lib/SPIRV/SPIRVWriter.cpp (hunk 1 above)
Comment (pauzinl) [reply lib/SPIRV/SPIRVWriter.cpp]: Thanks! Fixed it

**Review by MrSidims — APPROVED** [review]
LGTM
Will merge after @KornevNikita approval

**Review by KornevNikita — APPROVED** [review]
Thanks! Also LGTM!

**Pull request merged**